e2e/run-okd-tests.sh

```diff
@@ -10,63 +10,86 @@
+  ./minishift start --memory=4096 && eval $(./minishift oc-env)
+  oc login -u system:admin
+  oc adm policy add-cluster-role-to-user cluster-admin developer && oc login -u developer -p developer
+}
 
-# download oc client binary
-echo "[INFO] Download oc client"
-mkdir -p ${OPERATOR_REPO}/tmp
-chmod -R 777 ${OPERATOR_REPO}/tmp
-curl -s -S -L https://github.com/openshift/origin/releases/download/${OC_VERSION%%-*}/openshift-origin-client-tools-${OC_VERSION}-linux-64bit.tar.gz \
-  -o ${OPERATOR_REPO}/tmp/oc.tar && tar -xvf ${OPERATOR_REPO}/tmp/oc.tar -C ${OPERATOR_REPO}/tmp --strip-components=1
+oc_tls_mode() {
+    # generate self signed cert
+    echo "[INFO] Generate self signed certificate"
+    openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -subj "/CN=*.${IP_ADDRESS}.nip.io" -nodes && cat cert.pem key.pem > ca.crt
+    # replace default router cert
+    echo "[INFO] Update OpenShift router tls secret"
+    oc project default
+    oc secrets new router-certs tls.crt=ca.crt tls.key=key.pem -o json --type='kubernetes.io/tls' --confirm | oc replace -f -
+    echo "[INFO] Initiate a new router deployment"
+    sleep 20
+    oc rollout latest dc/router -n=default || true
+}
 
-# start OKD
-echo "[INFO] Start OKD ${OC_VERSION}"
-cd ${OPERATOR_REPO}/tmp
-rm -rf openshift.local.clusterup
-./oc cluster up --public-hostname=${IP_ADDRESS} --routing-suffix=${IP_ADDRESS}.nip.io
-./oc login -u system:admin
-./oc adm policy add-cluster-role-to-user cluster-admin developer
-./oc login -u developer -p password
-sleep 10
-echo "[INFO] Register a custom resource definition"
-./oc apply -f ${OPERATOR_REPO}/deploy/crds/org_v1_che_crd.yaml
+run_tests() {
+  minishift_installation
+  
+  echo "[INFO] Register a custom resource definition"
+  oc apply -f ${OPERATOR_REPO}/deploy/crds/org_v1_che_crd.yaml
 
-# generate self signed cert
-echo "[INFO] Generate self signed certificate"
-openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 365 -subj "/CN=*.${IP_ADDRESS}.nip.io" -nodes
-cat cert.pem key.pem > ca.crt
+  oc_tls_mode
+  
+  echo "[INFO] Compile tests binary"
+  docker run -t \
+              -v ${OPERATOR_REPO}/tmp:/operator \
+              -v ${OPERATOR_REPO}:/opt/app-root/src/go/src/github.com/eclipse/che-operator registry.access.redhat.com/devtools/go-toolset-rhel7:${GO_TOOLSET_VERSION} \
+              sh -c "OOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o /operator/run-tests /opt/app-root/src/go/src/github.com/eclipse/che-operator/e2e/*.go"
+  
+  echo "[INFO] Build operator docker image and load in to minishift VM..."
+  cd ${OPERATOR_REPO} && docker build -t che/operator -f Dockerfile . && docker save che/operator > operator.tar
+  eval $(./tmp/minishift docker-env) && docker load -i operator.tar && rm operator.tar
+  
+  echo "[INFO] Run tests..."
+  ./tmp/run-tests
+  
+  echo "[INFO] Deleting minishift VM..."
+  yes | ./tmp/minishift delete && rm -rf ~/.minishift ${OPERATOR_REPO}/tmp
```

Comment:
Since `set -e` when any command fails then minishift won't be deleted.
Is it better to move this to exit trap?

Reply:
Yeah. Your right if any commands fails before  `./tmp/minishift delete` minishift will not delete.

Reply:
It is better to keep `set -e` and delete minishift using bash exit trap

Reply:
Ok. I keep `set -e` and I add bash exit trap.